### PR TITLE
storage: read debezium transaction metadata from its persist shard

### PR DESCRIPTION
**Pull request by petrosagg**

### CODE REVIEW

File: src/coord/src/coord.rs [hunk 1]
```diff
@@ -651,7 +672,12 @@ impl<S: Append + 'static> Coordinator<S> {
                     )
                     .await?;
                 }
-                _ => (), // Handled in prior loop.
+                // Nothing to do for these cases
```
Comment (petrosagg) [inline src/coord/src/coord.rs:675]: @benesch Can you confirm this is the correct thing to do here?

Comment (benesch) [reply src/coord/src/coord.rs:679]: Yep, looks good!

File: src/storage/src/storage_state.rs [hunk 2]
```diff
@@ -1,6 +1,6 @@
 // Copyright Materialize, Inc. and contributors. All rights reserved.
 //
-// Use of this software is governed by the Business Source License
+// Use of this software is governed by the Business Scurce License
```
Comment (benesch) [inline src/storage/src/storage_state.rs:3]: Typo

**Review by benesch — APPROVED** [review]
This looks great! Thank yoU!

**Pull request merged**